compiler/src/iree/compiler/Codegen/Common/GPU/GPUCheckResourceUsage.cpp

```diff
@@ -57,29 +82,24 @@ static LogicalResult checkGPUAllocationSize(
   if (funcOp.getFunctionBody().empty())
     return success();
 
-  SmallVector<memref::AllocOp> allocOps;
+  SmallVector<Operation *> allocOps;
```

Comment:
Nit: Instead of two walks, you could do a single walk and make the callback look at any `Operation *` and there add it to the list if `isa<memref::AllocOp, memref::AllocaOp>`